orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -789,9 +789,9 @@ def test_changes_input_requires_grad_reinitializes_module_gradient_graph_builder
     model = ORTModule(model)
     x = torch.randn(N, D_in, device=device, requires_grad=True)
     model(x.data)
-    module_gradient_graph_builder = model._module_gradient_graph_builder
+    module_gradient_graph_builder_training = model._module_gradient_graph_builder_training
     model(x)
-    assert module_gradient_graph_builder != model._module_gradient_graph_builder
+    assert module_gradient_graph_builder_training != model._module_gradient_graph_builder_training
```

Comment:
Nit: This assertion leaves open the possibility that the two states are the inverse of what they should be :).

How about changing it as follows?

```
model(x.data)
assert not model._module_gradient_graph_builder_training
model(x)
assert model._module_gradient_graph_builder_training
```

It might be worth adding an assertion about the presence/absence of `grad_fn` on the result tensors from each call as well.

Reply:
I renamed the ```_module_gradient_graph_builder``` attribute to ```_module_gradient_graph_builder_training``` so as to distinguish it from the one used for inference in ```ORTModule```.
The change in this test is only reflective of the name changes. No other change was made.

The idea in this test is to just assert that when the input changes it requires_grad attribute from one forward call to the next, the ```module_gradient_graph_builder_training``` is reinstantiated.

I can add a check that shows that the after the ```backward()``` call, ```x.grad``` is ```None``` when the input x has ```requires_grad``` as ```False``` and not ```None``` when ```True```